### PR TITLE
Fix comparison against pd.Series

### LINKED ISSUE #4596: dask.dataframe.fillna fails with "ValueError: cannot reindex from a duplicate axis"
Starting with Dask 1.1.0, `dask.dataframe.fillna` fails when trying to fill based on a series from the same dataframe if the index is not unique. The following example works under Dask 1.0.0, but fails with more recent versions:

```
import dask
import dask.dataframe as dd 
import numpy as np 
import pandas as pd


@dask.delayed 
def f(): 
    return pd.DataFrame(dict(a=[1.0], b=[np.NaN])) 

ddf=dd.from_delayed([f(), f()], meta=dict(a=float, b=float))  
ddf.b=ddf.b.fillna(ddf.a)
ddf.compute()
```

### PR DESCRIPTION
**Pull request by amerkel2**
fixes #4596 

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
We switched to using `is_series_like` in order to support objects that looked like Pandas objects, but were not like cudf or, in some future, arrow.

**Comment by mrocklin** [comment]
The underlying issue may be resolved in https://github.com/dask/dask/pull/4533

**Comment by amerkel2** [comment]
No, even with #4533 merged the test added in this PR fails. The comment two lines above my change suggests that in this place, a different code path shall be taken for dask Series than for pandas series and similar. But `is_series_like` evaluates to `True` both for dask and pandas series, so it doesn't seem suitable here.

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ def fillna(self, value=None, method=None, limit=None, axis=None):
             # Control whether or not dask's partition alignment happens.
             # We don't want for a pandas Series.
             # We do want it for a dask Series
-            if is_series_like(value):
+            if isinstance(value, pd.Series):
```
Comment (mrocklin) [inline dask/dataframe/core.py]:
```suggestion
            if is_series_like(value) and not is_dask_collection(value):
```

How about this then?

Comment (amerkel2) [reply dask/dataframe/core.py]: Thanks, this works!

**Comment by mrocklin** [comment]
Thanks @amerkel2 .  This is in!

Also, I notice that this is your first code contribution to this repository.  Welcome!

**Comment by amerkel2** [comment]
Thank you, it's a pleasure!

**Pull request merged**